fuel/transformers/__init__.py

```diff
@@ -97,8 +114,9 @@ class Mapping(Transformer):
         data under source names `add_sources`.
 
     """
-    def __init__(self, data_stream, mapping, add_sources=None):
-        super(Mapping, self).__init__(data_stream)
+    def __init__(self, data_stream, produces_examples, mapping,
```

Comment:
It seems like a pretty fringe use case to be having a `Mapping` that has a different value of `produces_examples` than its data stream, don't you think? What about getting rid of the constructor argument and just doing

```python
@property
def produces_examples(self):
    return self.data_stream.produces_examples
```

(if you go this route, here it would be good to double check that it's not passed as kwargs, as it would have no effect)